src/services/services.ts

```diff
@@ -7530,7 +7533,7 @@ namespace ts {
         function walk(node: Node) {
             switch (node.kind) {
                 case SyntaxKind.Identifier:
-                    nameTable[(<Identifier>node).text] = (<Identifier>node).text;
+                    nameTable[(<Identifier>node).text] = nameTable[(<Identifier>node).text] === undefined ? node.pos : -1;

```

Comment:
Ah, okay. I misunderstood.

Why don't you use an enum that would make this explicit?

```ts
const name = (node as Identifier).text;
nameTable[name] = hasProperty(nameTable, name) ? NameTableState.Once : NameTableState.Many;
```

or at the very least, leave a nice big comment explaining what you're doing here and what `-1` means.